scripts/publish.js

```diff
@@ -18,44 +27,117 @@ function shouldUpdate(component, folder = COMPONENTS_PATH) {
     if (!packageJSON) return false;
 
     const name = packageJSON.name;
-
     // if a component doesn't exist in the registry then it must be published
     if (!JSON.parse(execSync(`npm search ${name} --json`, { encoding: 'utf8' })).length) return true;
 
     const localVersion = packageJSON.version;
     const publicVersion = getPublicVersion(name);
 
     if (localVersion !== publicVersion) {
-        console.log(`Package ${component} has new version - ${localVersion}, current is ${publicVersion}.`);
+        console.log(`Package ${component} has new local version - ${localVersion}. The current npm version is ${publicVersion}.`);
         return true;
     }
 
     return false;
 }
 
+/**
+ * Will generate the release notes based on the latest tag and the current one
+ * @param {string} tagName
+ * @returns {Promise<any>}
+ */
+async function getReleaseNotes(tagName) {
+    console.log(`Generating release notes for '${tagName}' based on the latest tag in the repo - ${latestTag}`);
+
+    return await octokit.request('POST /repos/{owner}/{repo}/releases/generate-notes', {
+        owner,
+        repo,
+        tag_name: tagName,
+        target_commitish: 'master',
+        previous_tag_name: latestTag,
+        configuration_file_path: '.github/release.yml',
+        headers: { 'X-GitHub-Api-Version': '2022-11-28' },
+    });
+}
+
+/**
+ * Will generate new release with a specified tag
+ * @param {string} tag
+ * @returns {undefined}
+ */
+async function createRelease(tag) {
+    try {
+        const release = await octokit.request(`GET /repos/{owner}/{repo}/releases/tags/{tag}`, {
+            owner,
+            repo,
+            tag: tag,
+            headers: { 'X-GitHub-Api-Version': '2022-11-28' },

```

Comment:
What happens if **octokit.request** fails because the version used in `X-GitHub-Api-Version` is no longer supported at some point?
![image](https://github.com/CoherentLabs/GameUIComponents/assets/21058220/d00d778d-b274-4881-bb20-b2fd10d8dbb6)

The question is more informative than anything because this version is probably going to be supported for a long time but in case it changes, are there going to be any consequences from that?

Reply:
The action will fail because the publish.js will exit with error code -1 and then we need to check why the action didn't pass. Then we know the version expired and we need to change it to a newer one.

I don't know any other way of handling this. If you have an idea share it with us.